implementations/rust/ockam/ockam_app_lib/src/incoming_services/state.rs

```diff
@@ -0,0 +1,365 @@
+                expires_at: "2020-09-12T15:07:14.00".to_string(),
+                grant_role: RoleInShare::Admin,
+                owner_email: "owner_email".to_string(),
+                scope: ShareScope::Project,
+                target_id: "target_id".to_string(),
+                ignored: false,
+            },
+            service_access_details: None,
+        };
+
+        // Skipping an invitation without service access details
+        let app_state = AppState::new(None, None, Some(CliState::test().unwrap())).unwrap();
+
+        app_state.context().runtime().block_on(async move {
+            app_state
+                .load_services_from_invites(vec![invitation.clone()])
+                .await;
+
+            let services = app_state.incoming_services().read().await.services.clone();
+            assert!(services.is_empty(), "No services should be loaded");
+
+            invitation.service_access_details = Some(ServiceAccessDetails {
+                project_identity: "I1234561234561234561234561234561234561234"
+                    .try_into()
+                    .unwrap(),
+                project_route: "mock_project_route".to_string(),
+                project_authority_identity: "Iabcdefabcdefabcdefabcdefabcdefabcdefabcd"
+                    .try_into()
+                    .unwrap(),
+                project_authority_route: "project_authority_route".to_string(),
+                shared_node_identity: "I12ab34cd56ef12ab34cd56ef12ab34cd56ef12ab"
+                    .try_into()
+                    .unwrap(),
+                shared_node_route: "remote_service_name".to_string(),
+                enrollment_ticket: EnrollmentTicket::new(
+                    OneTimeCode::new(),
+                    Some(ProjectLookup {
+                        node_route: None,
+                        id: "project_id".to_string(),
+                        name: "project_name".to_string(),
+                        identity_id: Some(
+                            Identifier::from_str("I1234561234561234561234561234561234561234")
+                                .unwrap(),
+                        ),
+                        authority: None,
+                        okta: None,
+                    }),
+                    None,
+                )
+                .hex_encoded()
+                .unwrap(),
+            });
+
+            app_state
+                .load_services_from_invites(vec![invitation.clone()])
+                .await;
+            let services = app_state.incoming_services().read().await.services.clone();
+            assert_eq!(1, services.len(), "One service should be loaded");
+
+            let service = &services[0];
+            assert_eq!(
+                "invitation_id",
+                service.id()
+            );
+            assert_eq!(
+                "I12ab34cd56ef12ab34cd56ef12ab34cd56ef12ab",
+                service.access_details.shared_node_identity.to_string()
```

Comment:
I think that this test breaks encapsulation. The behaviour we want to test is fully covered by L.322 and L324

Reply:
It may be superfluous, but I don't understand why would it break the encapsulation?

Reply:
That's the because the internal structure of service is accessed directly: `access_details.shared_node_identity`.
If we want to evolve the structure of the service, that's one more place where we need to change code.